### PR TITLE
Fixes #21

### LINKED ISSUE #21: "Upload to code-scanning" failure
Hi,

I am having an issue with the "Upload to code-scanning" step failing for my private repo with the following error,

```
Warning: Caught an exception while gathering information for telemetry: HttpError: Resource not accessible by integration. Will skip sending status report.
Error: Resource not accessible by integration
```

This appears to be related to [this issue](https://github.com/github/codeql-action/issues/2117) in codeql-action.

Adding the `actions: read` permission fixed it locally for me

### PR DESCRIPTION
**Pull request by paul-doherty**
Add action read permission to add codeql-action issue

### CODE REVIEW

File: .github/workflows/osv-scanner-unified-workflow.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ on:
     branches: ["main"]
 
 permissions:
+  # Required to upload SARIF file to CodeQL
+  actions: read
```
Comment (another-rex) [inline .github/workflows/osv-scanner-unified-workflow.yml:29]: Can you link to the issue you posted as well here? We'll want to remove this once that fix is merged in.

File: .github/workflows/osv-scanner-reusable.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@
 name: OSV-Scanner single vulnerability scan
 
 permissions:
+  actions: read
```
Comment (another-rex) [inline .github/workflows/osv-scanner-reusable.yml:19]: Add the same comment here and in osv-scanner-reusable-pr.yml as well.

**Review by another-rex — APPROVED** [review]
Thanks! Just a minor comment change, otherwise LGTM.

File: .github/workflows/osv-scanner-reusable.yml (hunk 2 above)
Comment (paul-doherty) [reply .github/workflows/osv-scanner-reusable.yml:20]: added the comment

File: .github/workflows/osv-scanner-unified-workflow.yml (hunk 1 above)
Comment (paul-doherty) [reply .github/workflows/osv-scanner-unified-workflow.yml:29]: done

**Pull request merged**